metrics_api/lib/opentelemetry/internal/proxy_instrument.rb

```diff
@@ -8,19 +8,22 @@ module OpenTelemetry
   module Internal
     # @api private
     class ProxyInstrument
-      def initialize(kind, name, unit, desc, callable)
+      def initialize(kind, name, unit, desc, callable, advice = nil)
```

Comment:
I'm not sure what the ideal default argument is. We expect advice to be a hash when it's present, but that might cause a lot of unnecessary object allocations. 

Right now, the API defaults to nil and the SDK defaults to an empty hash.

Reply:
Would a constant, empty, frozen hash be an option?

Reply:
Good idea! Updated in d2318b0da04dfea617ff52911ddc6deb760bfe00